doc/src/manual/code-loading.md

```diff
@@ -190,26 +205,32 @@ paths = Dict{Tuple{UUID,Symbol},String}(
 
 1. The private `Priv` package is "[vendored](https://stackoverflow.com/a/35109534/659248)" inside the `App` repository.
 2. The public `Priv` and `Zebra` packages are in the system depot, where packages installed and managed by the system administrator live. These are available to all users on the system.
 3. The `Pub` package is in the user depot, where packages installed by the user live. These are only available to the user who installed them.
 
+
 ### Package directories
 
-Package directories provide a kind of environment that approximates package loading in Julia 0.6 and earlier, and which resembles package loading in many other dynamic languages. The set of packages available in a package directory corresponds to the set of subdirectories it contains that look like packages: if `X/src/X.jl` is a file in a package directory, then `X` is considered to be a package and `X/src/X.jl` is the file Julia loads to get `X`. Which packages can "see" each other as dependencies depends on whether they contain project files, and if they do, on what appears in those project files' `[deps]` sections.
+Package directories provide a simpler environment, but less powerful or controlled, and they exist in two flavors (depending on whether they contain a project file). The set of root packages available is defined simply as the set of files in the directory that "look like" packages. A package `Name` is determined to exist if it finds one of the following files:
+  - A file `Name.jl`
+  - A file `Name/src/Name.jl`
+  - A file `Name.jl/src/Name.jl`
+Additionally, which packages it is able to import as dependencies depends on whether it contains a project file. If it does not, it can import any root package. Otherwise, if it does contain a `Package.toml` file, it can only import those packages which are identified in its `[deps]` sections (by uuid and name).
 
-**The roots map** is determined by the subdirectories `X` of a package directory for which `X/src/X.jl` exists and whether `X/Project.toml` exists and has a top-level `uuid` entry. Specifically `:X => uuid` goes in `roots` for each such `X` where `uuid` is defined as:
+**The roots map** is determined by enumerating the contents of the package directory to generate a list of all packages that exist.
+Additionally, a UUID will be assigned to each entry as follows: For a given package found inside the folder `Name`...
 
-1. If `X/Project.toml` exists and has a `uuid` entry, then `uuid` is that value.
-2. If `X/Project.toml` exists and but does *not* have a top-level UUID entry, `uuid` is a dummy UUID generated by hashing the canonical path of `X/Project.toml`.
-3. If `X/Project.toml` does not exist, then `uuid` is the all-zero [nil UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier#Nil_UUID).
+1. If `Name/Project.toml` exists and has a `uuid` entry, then `uuid` is that value.
```

Comment:
I think using the name `X` consistently throughout the document was clearer. Now we use `X` early on and then switch to `Name` later which seem more likely to be an actual package name and less clearly a placeholder.

Reply:
Yep, I'll switch back. Was just interesting in seeing whether it felt better or not. Either way, we would certainly need to be more consistent about the change if we did decide that a longer name (`Name`, `Xray`, `FooBar`, or whatever) was clearer (defined as simpler to read and more obvious that each instance of X must be the same string, and not just a unique arbitrary placeholder).